R/slice_derivation.R

```diff
@@ -83,20 +84,44 @@
 #'     args = params(time_imputation = "last")
 #'   )
 #' )
+#'
 slice_derivation <- function(dataset,
                              derivation,
                              args = NULL,
                              ...) {
   # check input
   assert_data_frame(dataset)
-  assert_function(derivation, params = c("dataset"))
+  assert_function(derivation)
   assert_s3_class(args, "params", optional = TRUE)
   if (!is.null(args)) {
     assert_function(derivation, names(args))
   }
   slices <- list2(...)
   assert_list_of(slices, "derivation_slice")
 
+  # Check that every mandatory argument to the derivation function is either passed
+  # inside args or present in every slice
+  mandatory_args <- formals(derivation) %>% keep(is_missing) %>% names()
+
+  if(length(mandatory_args > 1)){
+    mandatory_args <- mandatory_args[-1] # ignore first item, as that is the dataset
+
+    for (mandatory_arg in mandatory_args){
+
+      if (!mandatory_arg %in% names(args)){
+
+        check <- sapply(slices, function(x) mandatory_arg %in% x)
+
+        if(!all(check)){
+          cli_abort(
+            "The mandatory argument `{mandatory_arg}` of derivation function `{deparse(substitute(derivation))}`
+            has either not been passed to the `args` argument, or is not present in all derivation slices."
```

Comment:
Shouldn't it be
```suggestion
            has neither been passed to the `args` argument, nor is present in all derivation slices."
```
?

Reply:
**neither** has ~neither~ been passed to the `args` argument, nor is present in all derivation slices.

Reply:
I get
```
Issue with the mandatory argument `mode` of derivation function `derive_var_extreme_flag`. It has either: (1) not been
  passed to the `args` argument, or (2) not been passed to all derivation slices.
```
now.
However, to me it sounds like the mandatory argument has to be specified for both the `args` argument _and_ all derivation slices.

Reply:
Maybe it is clearer if we state what it expected, e.g.,
```
Issue with the mandatory argument `mode` of derivation function `derive_var_extreme_flag`.
It must: (1) been passed to the `args` argument, or (2) been passed to all derivation slices.
```
What do you think?

Reply:
Yes, that's better thanks